app/register/search.js

```diff
@@ -0,0 +1,117 @@
+const sequelize = require('../config/db')
+const { Op } = require('sequelize')
+const { sortAndGroupResults } = require('./search/sorting-and-grouping')
+const { cleanupSearchTerms } = require('./search/search-terms')
+const { mapResults } = require('./search/search-results')
+const { fuzzySearch } = require('../repos/search-match-codes')
+const { trigramSearch } = require('../repos/search-tgrams')
+const { rankResult } = require('../repos/search-rank')
+const { buildTsVectorQuery } = require('./search/search-builder')
+
+const trigramQueryThreshold = 0.4
+const trigramRankThreshold = 1.001
+const fuzzyRankThreshold = 1.001
+const fullTextRankThreshold = 1.01
+
+const rankAndKeep = (results, terms, threshold, type) => {
+  const toKeep = []
+
+  const numRecords = results.length
+
+  results.forEach(res => {
+    res.rank = rankResult(terms, res, type)
+    if (res.rank >= threshold || numRecords < 11) {
+      toKeep.push(res)
+    }
+  })
+
```

Comment:
this function will have side-effects. It would be better to do a map then a filter. Something like this:

`results.map(result => ({ ...result, rank: rankResult(terms, res, type) })).filter(res => res.rank >= threshold || numRecords < 11)`